### DISCUSSION TITLE
Integration of Real-Time Sonar-Based Map Drawing

### DISCUSSION THREAD

**Discussion by SHengZi578** [FakeFishGames/Barotrauma#14767] [Ideas and suggestions]
Currently, the game features a sonar device that provides real-time detection of the player's surroundings. To build upon this, I suggest implementing a mapping system that allows players to use the sonar data to dynamically draw and update a larger, more comprehensive map of the underwater environment. This would enable players to not only navigate more effectively but also gain a better understanding of the terrain, resource distribution, and potential hazards. The ability to create detailed maps based on sonar scans would add a new layer of depth to the gameplay, fostering a greater sense of exploration and discovery. I believe this feature would greatly enrich the "Subnautica" experience and resonate positively with the player community.

**Comment by Ek-Videogames** [comment]
Would also be useful in maps whos width of tunnel is larger than the sub's sonar range making it difficult to tell where the intended path is at a glance. Also no doubt useful for particularly long cave systems too if implemented on the handheld sonar as well.

Some button that displays a 'full screen map' perhaps.

Once 'detected', display the map Voronoi geometry edge with simple vector lines?
